File: NeetCode/Python/LinkedList/Easy/206.ReverseLinkedList.py
# Given the head of a singly linked list, reverse the list, 
# and return the reversed list.

class ListNode(object):
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


def reverseList(head):
    current = head
    previous = None

    while current:
        next_node = current.next

        current.next = previous

        previous = current
        current = next_node

    return previous

#[1,2,3,4,5]

File: swift/Neetcode/LinkedList/Easy/206.ReverseLinkedList.swift
// Given the head of a singly linked list, reverse the list, 
// and return the reversed list.

func reverseList(_ head: ListNode?) -> ListNode? {
    var previous: ListNode? = nil
    var current = head

    while current !== nil {
        let next = current?.next
        current?.next = previous
        previous = current
        current = next
    }

    return previous
}